py3status/modules/updates.py

```diff
@@ -0,0 +1,215 @@
+        if "No packages to upgrade." in output:
+            return ""
+        return output
+
+
+class Zypper(Update):
+    def count_updates(self, output):
+        return len([x for x in output.splitlines() if x][4:])
+
+
+class Py3status:
+    """
+    """
+
+    # available configuration parameters
+    cache_timeout = 600
+    format = None
+    thresholds = [
+        (0, 'darkgray'), (10, 'degraded'), (20, 'orange'), (30, 'bad')]
+
+    class Meta:
+        deprecated = {
+            "rename_placeholder": [
+                {
+                    "placeholder": "aur",
+                    "new": "auracle",
+                    "format_strings": ["format"],
+                }
+            ]
+        }
+
+    def post_config_hook(self):
+        managers = update_managers = [
+            ("Pacman", "checkupdates"),
+            ("Auracle", "auracle sync --color=never"),
+            ("Yay", "yay --query --upgrades --aur"),
+            ("Apk", "apk version -l '<'"),
+            ("Apt", "apt list --upgradeable"),
+            ("Eopkg", "eopkg list-upgrades"),
+            ("Pkg", "pkg upgrade --dry-run --quiet"),
+            ("Xbps", "xbps-install --update --dry-run"),
+            ("Zypper", "zypper list-updates"),
+        ]
+
+        managed = getattr(self, 'managers', None)
+        if managed:
+            new_managers = []
+            for custom in managed:
+                if isinstance(custom, tuple):
+                    if len(custom) != 2 or custom[0].lower() == 'update':
+                        raise Exception(STRING_INVALID)
+                    new_managers.append(custom)
+                else:
+                    name = custom.lower()
+                    if name == 'update':
+                        raise Exception(STRING_INVALID)
+                    for manager in update_managers:
+                        if manager[0].lower() == name:
+                            new_managers.append(manager)
+                            break
+            managers = new_managers
+
+        placeholders = self.py3.get_placeholders_list(self.format or '')
+        placeholders = [x for x in placeholders if x != 'update']
+
+        formats = []
+        self.backends = []
+        for name, command in managers:
+            name_lowercased = name.lower()
+            if placeholders:
+                for placeholder in placeholders:
+                    if placeholder == name_lowercased:
+                        break
+                else:
+                    continue
+            if self.py3.check_commands(command.split()[0]):
+                try:
+                    backend = globals()[name.capitalize()]
+                except KeyError:
+                    backend = Update
+                formats.append((name, name_lowercased))
+                self.backends.append(backend(self, name_lowercased, command))
+
+        if not self.format:
+            auto = "[\?not_zero {name} [\?color={manager} {{{manager}}}]]"
```

Comment:
what does `{manager}` placeholder contain, number of available updates? Should we rename `{name}` to `{manager}` and `{manager}` to `{count}` ?

Reply:
I'll rename it to `{name}`, `{lower}`. The `{name}` is what we put in... Custom names. `{lower}` is same (lowercased) for color options and placeholders.

> format = "[[\?not_zero PACKage [\?color=package {package}]|\?show No Updates]"